SwiftyJSONAPI/JSONAPIResource.swift

```diff
@@ -92,7 +95,7 @@ public class JSONAPIResource: JSONPrinter {
         return attributes[key]
     }
     
-    func loadResources(withIncludedResources includedResources: ResourcesByTypeAndId) {
+    func loadResources(withIncludedResources includedResources: ResourcesByTypeAndId, cachedResources: inout CachedResources) {
```

Comment:
One possible improvement to the API would be:
```
func loadResources(withIncludedResources includedResources: ResourcesByTypeAndId) {
    func _loadResources(withIncludedResources includedResources: ResourcesByTypeAndId, cachedResources: inout CachedResources) { <SAME IMPLEMENTATION>}

    var cachedResources = CachedResources()
    _loadResources(withIncludedResources: includedResources, cachedResources: &cachedResources)    
}
```

This avoids having to provide an empty Set from JSONAPIDocument.

Reply:
I do like the approach but I need to recursively call loadResources on the resource itself which will require passing the cache in:
https://github.com/thomassnielsen/SwiftyJSONAPI/blob/017ad7911bef410d6af9227383160d574a186d19/SwiftyJSONAPI/JSONAPIResource.swift#L113

Reply:
Ahh I see, makes sense! I will make the change, thanks @alessandro-martin 😄